rust-runtime/aws-smithy-types/src/instant/format.rs

```diff
@@ -27,18 +27,69 @@ impl fmt::Display for DateParseError {
     }
 }
 
-pub(crate) mod http_date {
+fn remove_trailing_zeros(string: &mut String) {
+    while let Some(b'0') = string.as_bytes().last() {
+        string.pop();
+    }
+}
+
+pub(crate) mod epoch_seconds {
+    use super::remove_trailing_zeros;
+    use super::DateParseError;
+    use crate::Instant;
     use std::str::FromStr;
 
-    use chrono::{Datelike, NaiveDate, NaiveDateTime, NaiveTime, Timelike, Weekday};
+    /// Formats an `Instant` into the Smithy epoch seconds date-time format.
+    pub(crate) fn format(instant: &Instant) -> String {
+        if instant.subsecond_nanos == 0 {
+            format!("{}", instant.seconds)
+        } else {
+            let mut result = format!("{}.{:0>9}", instant.seconds, instant.subsecond_nanos);
+            remove_trailing_zeros(&mut result);
+            result
+        }
+    }
 
+    /// Parses the Smithy epoch seconds date-time format into an `Instant`.
+    pub(crate) fn parse(value: &str) -> Result<Instant, DateParseError> {
+        let mut parts = value.split('.');
+        let (mut whole, mut decimal) = (0i64, 0u32);
+        if let Some(whole_str) = parts.next() {
+            whole = <i64>::from_str(whole_str).map_err(|_| DateParseError::IntParseError)?;
+        }
+        if let Some(decimal_str) = parts.next() {
+            // Pad the decimal str component out to 9 characters on the stack
+            let mut decimal_buf = [b'0'; 9];
+            for (i, b) in decimal_str.as_bytes().iter().enumerate() {
+                if i >= 9 {
+                    return Err(DateParseError::Invalid("decimal is too long"));
+                }
+                if *b < b'0' || *b > b'9' {
+                    return Err(DateParseError::Invalid("decimal must be alphanumeric"));
+                }
+                decimal_buf[i] = *b;
+            }
+            // Unsafe: the decimal_buf is only populated with ASCII numeric characters above
+            debug_assert!(std::str::from_utf8(&decimal_buf).is_ok());
```

Comment:
we can avoid unsafe here but just counting how many zeros need to be added and performing a multiplication on the final result, right?

Reply:
Yep. My debate was between this or a for loop that multiplies by 10. Which is uglier?

Reply:
I'd do the for loop that multiplies by 10, personally. No need for any unsafe or any custom input validation, just multiplying by 10 according to the length of the string.

the wrinkle in this case is trimming spaces from the string